accountClient/api.go

```diff
@@ -81,6 +101,24 @@ type ValidateTokenResponse struct {
 	Valid     bool   `json:"valid"`      //Whether the token was valid
 }
 
+// CreateRegTokenRequest represents a valid request to the account service's /tokens endpoint POST.
+type CreateRegTokenRequest struct {
+	TokenPermissions
+}
+
+// CreateRegTokenResponse  represents the result of calling the account service's /tokens POST endpoint.
+type CreateRegTokenResponse struct {
+	Token       string           `json:"token"`
+	Permissions TokenPermissions `json:"permissions"`
+}
+
+// TokenPermissions permissions associated with a registration token.
+// called ClientPermissions in the account service spec
+type TokenPermissions struct {
+	Enabled bool `json:"enabled"`  // Flag a newly created client as enabled even if the default behavior is creating disabled clients
+	OneTime bool `json:"one-time"` // Automatically delete the token after it's been used to register a client
```

Comment:
snake case for JSON names

Reply:
I'll have to check, but I'm pretty sure I changed this to snake case in the account service refactor. I agree snake case is correct. It was in the API before, but unused, now it is used for the registration flow. Changing it is not any kind of a breaking change.

Reply:
Taken from the current spec, but since it's been changed I'll update it

Reply:
I do have it updated the account service to use snake-case. The original code was camel case, but agree, underscores for JSON names is better all around.

Reply:
updated 👍